### ISSUE TITLE
Drop full daily softlinking in favour of pruning latest

### ISSUE THREAD

**Issue by SamStudio8** [SamStudio8/elan-nextflow#12]
After Elan finishes, we burn a lot of cycles individually softlinking each FASTA, BAM and QC file that has passed basic QC from the start of COG until today. These folders are created each day and the most recent is softlinked as `latest`. It's the impression of the pipelines group that nobody is fetching FASTA or BAM by anything other than the `latest` directory.

My plan would be to replace the current process of linking all the individual files every day, with one where we only need to check what new softlinks are needed from today's run to be added into `latest`. It will still be possible to reach the individual FASTA and BAM through `latest` (as normal). This will also not change how we generate the daily merged consensus and metadata files. The dated directories will still exist, this plan would just mean the FASTA and BAM softlinks will no longer be added to them.

This would save at least 30-45 minutes of valuable tree-building/downstream analysis time which likely outweighs any reason to keep this method in place, however, as a means of providing compatibility for anyone who would want this, we can always produce a txt file of all the file paths that would have been softlinked on that day.

-- 20210127
* [x] Deprecate consensus `cat` and replace with `sbatch`'ed "matched" FASTA. Symlink so both FASTA are the same to stop broken scripts. Alternatively, sort the FASTA lexo; sort the metadata - and try best effort walk n' match.
* [x] No FASTA or BAM symlinks to be provided outside `latest/` dir going forward.

**Comment by SamStudio8** [comment]
Working on this to this now, especially as throughput has changed dramatically since October.

**Comment by SamStudio8** [comment]
Successfully deployed and tested today... 180 minute improvement ;) This was long overdue.

**Comment by SamStudio8** [comment]
Although this was a considerable improvement, the `cp -r` step to copy yesterday's symlinks from `latest` to today's `date` dir still takes some time, especially as we continue to accrue samples. It is the view of DIPI and the CLIMB system administrators that there is no good reason to continue to provide a mechanism for users to fetch single FASTA via symlink for each Elan `date` dir - especially when the sequences can be readily obtained for the corresponding date's merged FASTA. This simply costs too much time.

Tomorrow I will work to deprecate the historical symlink process. `latest/` will become a real directory, housing only the symlinks that reflect the COG-UK data set as it stands. The `latest` dir symlinks will be pruned in the same way as the publishing script currently prunes the dated symlinks directory. Dated dirs will now only hold the relevant logs, FASTA and TSV. Latest will symlink to those files as usual.

**Comment by SamStudio8** [comment]
https://github.com/SamStudio8/elan-nextflow/blob/master/bin/control/cog-publish.sh#L132 is an urgent candidate for swapping to an `sbatch` script.

**Comment by SamStudio8** [comment]
I've deprecated the "unmatched" `cat` FASTA. A symlink will bind the unmatched FASTA to be the matched one. This should have no adverse effect because users were unable to rely on the headers found in the unmatched FASTA anyway. Combined with moving the reconcile step to SLURM, I expect we'll save 90 minutes on average going forward.

**Comment by SamStudio8** [comment]
With these three changes I anticipate we'll save between 150 and 180 minutes today.

**Comment by SamStudio8** [comment]
Nailed this, other than failing to notice that the new reconcile batch script would need to point to `latest/` to be able to use the symlinks. As missing symlinks are an expected part of running the reconcile script, it happily skipped all the sequences (because the dated directories no longer contain any) without error, wrote an empty FASTA, exited 0 and happily sent out a completion message to `tael`.

I've killed `Asklepian` before it sends garbage to PHE. Luckily `phylopipe` wasn't listening at the time so we don't need to handle that. I have corrected the oversight (https://github.com/SamStudio8/elan-nextflow/commit/75c95aa52e2afcf6a6803f156fff15d75fa5c2d6) and am running the reconcile step separately now. Nothing else is at risk. I will raise `Asklepian` automatically by sending another `tael` message as soon as this is completed.
